src/jwks.cc

```diff
@@ -184,16 +182,24 @@ Status extractJwkFromJwkEC(const ::google::protobuf::Struct& jwk_pb,
     return Status::JwksECKeyBadCrv;
   }
   jwk->crv_ = crv_str;
+
+  // If both alg and crv specified, make sure they match
+  if (jwk->alg_specified_ && !jwk->crv_.empty()) {
+    if (!((jwk->alg_ == "ES256" && jwk->crv_ == "P-256") ||
+          (jwk->alg_ == "ES384" && jwk->crv_ == "P-384") ||
+          (jwk->alg_ == "ES512" && jwk->crv_ == "P-521"))) {
+      return Status::JwksECKeyAlgNotCompatibleWithCrv;
+    }
+  }
+
   int nid;
-  if (jwk->alg_ == "ES256" && (jwk->crv_ == "P-256" || jwk->crv_ == "")) {
+  if (jwk->alg_ == "ES256" || jwk->crv_ == "P-256") {
     nid = NID_X9_62_prime256v1;
     jwk->crv_ = "P-256";
-  } else if (jwk->alg_ == "ES384" &&
-             (jwk->crv_ == "P-384" || jwk->crv_ == "")) {
+  } else if (jwk->alg_ == "ES384" || jwk->crv_ == "P-384") {
     nid = NID_secp384r1;
     jwk->crv_ = "P-384";
-  } else if (jwk->alg_ == "ES512" &&
-             (jwk->crv_ == "P-521" || jwk->crv_ == "")) {
+  } else if (jwk->alg_ == "ES512" || jwk->crv_ == "P-521") {
     nid = NID_secp521r1;
     jwk->crv_ = "P-521";
   } else {
```

Comment:
Can we now return a different error in this case? Maybe JwksECKeyAlgOrCrvUnsupported?

Reply:
Done